libs/designer/src/lib/ui/panel/configureTemplatePanel/configureWorkflowsPanel/tabs/selectWorkflowsTab.tsx

```diff
@@ -24,82 +46,219 @@ export const SelectWorkflows = () => {
+      id: 'mXwGRH',
+      description: 'Label for workflow trigger type',
+    }),
+  };
+
+  type WorkflowsTableItem = {
+    id: string;
+    name: string;
+    trigger: string;
+  };
+
+  const columns: TableColumnDefinition<WorkflowsTableItem>[] = [
+    createTableColumn<WorkflowsTableItem>({
+      columnId: 'name',
+    }),
+    createTableColumn<WorkflowsTableItem>({
+      columnId: 'trigger',
+    }),
+  ];
+
+  const items =
+    workflows?.map((workflow) => ({
+      id: workflow.id,
+      name: workflow.name,
+      trigger: workflow.triggerType,
+    })) ?? [];
+
+  const {
+    getRows,
+    selection: { allRowsSelected, someRowsSelected, toggleAllRows, toggleRow, isRowSelected },
+  } = useTableFeatures(
+    {
+      columns,
+      items,
+    },
+    [
+      useTableSelection({
+        selectionMode: 'multiselect',
+        selectedItems: new Set(Object.keys(selectedWorkflowsList)),
+        onSelectionChange: (_, data) => {
+          onWorkflowsSelected(Array.from(data.selectedItems, String).map((workflowsId) => normalizedWorkflowId(workflowsId)));
+        },
+      }),
+    ]
+  );
+
+  const rows = getRows((row) => {
+    const selected = isRowSelected(normalizedWorkflowId(row.item.id));
+    return {
+      ...row,
+      onClick: (e: React.MouseEvent) => toggleRow(e, normalizedWorkflowId(row.item.id)),
+      onKeyDown: (e: React.KeyboardEvent) => {
+        if (e.key === ' ') {
+          e.preventDefault();
+          toggleRow(e, normalizedWorkflowId(row.item.id));
+        }
+      },
+      selected,
+      appearance: selected ? ('brand' as const) : ('none' as const),
+    };
+  });
+
+  const toggleAllKeydown = useCallback(
+    (e: React.KeyboardEvent<HTMLDivElement>) => {
+      if (e.key === ' ') {
+        toggleAllRows(e);
+        e.preventDefault();
+      }
+    },
+    [toggleAllRows]
   );
 
   return (
-    <div>
-      <ResourcePicker viewMode={'alllogicapps'} onSelectApp={onLogicAppSelected} />
-      <div>
-        <h4>Workflows</h4>
-        <div>
-          {isLoading ? (
-            <div>Loading...</div>
-          ) : (
-            workflows?.map((workflow) => (
-              <Checkbox
-                onChange={(_, event) => onWorkflowSelected(workflow.id, !!event.checked)}
-                checked={hasProperty(workflowsInTemplate, workflow.id) || isConsumption}
-                disabled={isConsumption}
-                label={workflow.name}
-                key={workflow.id}
+    <div className="msla-templates-tab msla-panel-no-description-tab">
+      <TemplatesSection title={intlText.SOURCE} titleHtmlFor={'sourceLabel'} description={intlText.SOURCE_LABEL}>
+        <ResourcePicker viewMode={'alllogicapps'} onSelectApp={onLogicAppSelected} />
+      </TemplatesSection>
+      <TemplatesSection title={intlText.WORKFLOWS} titleHtmlFor={'workflowsLabel'} description={intlText.WORKFLOWS_LABEL}>
+        <Table aria-label="Table with multiselect" style={{ minWidth: '550px' }}>
```

Comment:
The aria label for tables should read, Workflows list table.. so during navigation screen readers reads it out correctly